src/main/java/io/qdrant/client/QdrantClient.java

```diff
@@ -665,6 +674,76 @@ public ListenableFuture<List<AliasDescription>> listAliasesAsync(@Nullable Durat
 
 	//endregion
 
+	//region ShardKey Management
+
+		/**
+	 * Creates a shard key for a collection.
+	 *
+	 * @param createShardKey The request object for the operation.
+	 * @return a new instance of {@link ListenableFuture}
+	 */
+	public ListenableFuture<CreateShardKeyResponse> createShardKeyAsync(CreateShardKeyRequest createShardKey) {
+		return createShardKeyAsync(createShardKey, null);
+	}
+
+	/**
+	 * Creates a shard key for a collection.
+	 *
+	 * @param createShardKey The request object for the operation.
+	 * @param timeout The timeout for the call.
+	 * @return a new instance of {@link ListenableFuture}
+	 */
+	public ListenableFuture<CreateShardKeyResponse> createShardKeyAsync(CreateShardKeyRequest createShardKey, @Nullable Duration timeout) {
+		String collectionName = createShardKey.getCollectionName();
+		Preconditions.checkArgument(!collectionName.isEmpty(), "Collection name must not be empty");
+		logger.debug("Create shard key'{}'", collectionName);
+
+		ListenableFuture<CreateShardKeyResponse> future = getCollections(timeout).createShardKey(createShardKey);
+		addLogFailureCallback(future, "Create shard key");
+		return Futures.transform(future, response -> {
+			if (!response.getResult()) {
+				logger.error("Shard key could not be created for '{}'", collectionName);
+				throw new QdrantException("Shard key could not be created for '" + collectionName);
+			}
+			return response;
+		}, MoreExecutors.directExecutor());
+	}
+
+	/**
+	 * Deletes a shard key for a collection.
+	 *
+	 * @param deleteShardKey The request object for the operation.
+	 * @return a new instance of {@link ListenableFuture}
+	 */
+	public ListenableFuture<DeleteShardKeyResponse> deleteShardKeyAsync(DeleteShardKeyRequest deleteShardKey) {
+		return deleteShardKeyAsync(deleteShardKey, null);
+	}
+
+	/**
+	 * Deletes a shard key for a collection.
+	 *
+	 * @param deleteShardKey The request object for the operation.
+	 * @param timeout The timeout for the call.
+	 * @return a new instance of {@link ListenableFuture}
+	 */
+	public ListenableFuture<DeleteShardKeyResponse> deleteShardKeyAsync(DeleteShardKeyRequest deleteShardKey, @Nullable Duration timeout) {
+		String collectionName = deleteShardKey.getCollectionName();
+		Preconditions.checkArgument(!collectionName.isEmpty(), "Collection name must not be empty");
+		logger.debug("Delete shard key'{}'", collectionName);
+
+		ListenableFuture<DeleteShardKeyResponse> future = getCollections(timeout).deleteShardKey(deleteShardKey);
+		addLogFailureCallback(future, "Delete shard key");
+		return Futures.transform(future, response -> {
+			if (!response.getResult()) {
+				logger.error("Shard key could not be deleted for '{}'", collectionName);
```

Comment:
log both shard key and collection name

```suggestion
				logger.error("Shard key '{}' could not be deleted for '{}'", shardKey, collectionName);
```